Newsletters: add 255 (2023-06-14)

harding opened the pull request:
- [ ] Lede, releases/RCs, topic links @harding
- [ ] Waiting for confirmation 5 @glozow 
- [ ] Bitcoin Core PR Review Club summary @LarryRuane 
- [ ] Bitcoin Core 27501 @adamjonas 
- [ ] Eclair 2677 @Xekyo (let me know if you want to pass on this LN-focused PR)

LarryRuane commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,176 @@
+    don't currently relay transactions that contain annexes, this is not
+    a problem at present.  Jager [suggested][jager annex3] that some
+    [previous work][bitcoin core #24007] in Bitcoin Core could possibly
+    address this problem
```
```suggestion
    address this problem.
```

harding commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,176 @@
+    a soft fork.
+
+    Greg Sanders [replied][sanders annex] to ask what data Jager
+    specifically wanted to store in the annex and described his own use
```
Sorry, this was written before the later discussion where it became clear to me that was the usecase you were really interested in.  I've updated the text to reference and link to your answer.  Thanks!

joostjager replied:
Sorry, I wasn't sure how final the text already was. Thanks for the update!

glozow commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,183 @@
+  q4="Suppose we pass a vector of 40 eviction candidates to `SelectNodeToEvict()`.
+      Before and after this PR, what’s the theoretical maximum of Tor nodes
+      that can be protected from eviction?"
+  a4="With and without the PR, the number would be 34 out of 40, assuming
```
nit, maybe:
```suggestion
  a4="With and without the PR, the number would be 34 out of 40 respectively, assuming
```

LarryRuane replied:
I don't quite see this; for me, you use "respectively" in a sentence structure like, "May and June have 31 and 30 days, respectively" (sort of like `zip()` in Python).

glozow commented on _includes/specials/policy/en/05-dos.md:
```diff
@@ -0,0 +1,124 @@
+and replacement transactions, the new transaction must pay "new" fees
+to cover the cost of its own bandwidth at 1sat/vB.
+
+This incremental relay feerate is also used to [raise][pr 6722] a
+node's mempool minimum feerate ("mempoolminfee") when it reaches
+capacity and evicts transactions in descendant score order: the new
+mempool minimum feerate is set to 1sat/vB higher than the descendant
+score of the last transaction (package) evicted, then decays downwards
+towards 1sat/vB at a rate adjusted for how quickly the mempool
+empties. Setting a dynamic mempool minimum feerate does not
+necessarily affect the outcome, as a transaction with a feerate below
+the just-evicted transaction would be evicted immediately after being
+accepted.  However, rejecting the transaction faster affords
+worthwhile savings in validation effort.
+
```
I know I'm over 1000 words... my eviction algorithm would kick this paragraph first, as maybe the least interesting example in the post.

Xekyo replied:
Yeah, that might be a bit too deep into the weeds, although eviction has been a topic in the past few months that might have people curious.

Xekyo commented on _includes/specials/policy/en/05-dos.md:
```diff
@@ -0,0 +1,124 @@
+one and adds the rest to a work queue to be processed after other
+peers' messages. A node also typically first downloads a block header
+and verifies its Proof of Work prior to downloading and validating the
+rest. Thus, any attacker wishing to exhaust this node's resources
```
“the rest” of what?

```suggestion
rest of the block. Thus, any attacker wishing to exhaust this node's resources
```

Xekyo commented on _includes/specials/policy/en/05-dos.md:
```diff
@@ -0,0 +1,124 @@
+thousands of inputs which took an abnormally long time to validate due
+to [signature verification and quadratic sighashing][rusty megatx]. An
+attacker may also make all but the last signature valid, causing the
+node to spend minutes on this transaction, only to find that it is
```
```suggestion
node to spend minutes on their transaction, only to find that it is
```

Xekyo commented on _includes/specials/policy/en/05-dos.md:
```diff
@@ -0,0 +1,124 @@
+to [signature verification and quadratic sighashing][rusty megatx]. An
+attacker may also make all but the last signature valid, causing the
+node to spend minutes on this transaction, only to find that it is
+garbage. During that time, a new block may have arrived but the node
+was too busy to process it. One can imagine this type of attack being
```
```suggestion
garbage. During the time it is busy processing the transaction, the node would delay processing a new block that is announced. One can imagine this type of attack being
```

Xekyo commented on _includes/specials/policy/en/05-dos.md:
```diff
@@ -0,0 +1,124 @@
+requirements. A non-mining node doesn’t ever receive fees – they are
+only paid to the miner who confirms the transaction – and some network
+bandwidth has already been used by the time the fee is checked.
+However, fees represent a cost to the attacker. Somebody may send an
+extremely high amount of transactions that require validation and
+relay, but will at some point run out of money to pay the fees.
```
```suggestion
However, fees represent a cost to the attacker. The idea is that somebody who sends
a lot of transactions that require validation and
relay will eventually run out of money to pay the fees.
```

Xekyo commented on _includes/specials/policy/en/05-dos.md:
```diff
@@ -0,0 +1,124 @@
+
+The Replace by Fee [policy implemented by Bitcoin Core][bitcoin core
+rbf docs] requires that the replacement transaction pay a higher
+feerate than the transaction(s) it directly conflicts with, but also
```
```suggestion
feerate than each transaction it directly conflicts with, but also
```

Xekyo commented on _includes/specials/policy/en/05-dos.md:
```diff
@@ -0,0 +1,124 @@
+The Replace by Fee [policy implemented by Bitcoin Core][bitcoin core
+rbf docs] requires that the replacement transaction pay a higher
+feerate than the transaction(s) it directly conflicts with, but also
+requires that it pay a higher total fee. The additional fees (the fees
```
```suggestion
requires that it pay a higher total fee than all transactions it displaces. The additional fees (the fees
```

Xekyo commented on _includes/specials/policy/en/05-dos.md:
```diff
@@ -0,0 +1,124 @@
+paid by the replacement subtracted by the total fees paid by all of
+the transactions it replaces) divided by the replacement transaction's
+virtual size must be at least 1sat/vB. This fee policy is not
+primarily concerned with the incentive compatibility. Rather, it
+prevents repeated replacements of the same transaction in which each
```
Maybe something along the lines of “Rather, this incurs a minimum cost for repeated transaction replacements to curb bandwidth wasting attacks”?

Xekyo commented on _includes/specials/policy/en/05-dos.md:
```diff
@@ -0,0 +1,124 @@
+
+A node that fully validates blocks and transactions requires resources
+including memory, computational resources, and network bandwidth. We
+must keep the requirements for these resources low in order to
```
```suggestion
must keep the resource requirements low to
```

Xekyo reviewed (commented):
I just looked at the “waiting for confirmation” article as it’s much later for Gloria already, other review to follow.

Xekyo commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,190 @@
+  state of all configuration options to restarted plugins.
+
+- [Eclair #2677][] increases the default `max_cltv` from 1,008 blocks
+  (about one week) to 2,016 blocks (about two weeks). This extends
```
```suggestion
  (about one week) to 2,016 blocks (about two weeks). This extends the
```

Xekyo commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,190 @@
+
+- [Eclair #2677][] increases the default `max_cltv` from 1,008 blocks
+  (about one week) to 2,016 blocks (about two weeks). This extends
+  maximum permitted number of blocks until a lightning payment attempt
```
```suggestion
  maximum permitted number of blocks until a payment attempt
```

Xekyo commented:
Looks good to me so far

LarryRuane reviewed (commented):
I love the policy write-up! Please never feel a need to justify not taking any of my suggestions; the last thing I want to do is require more work from anyone! Ignore anything you see differently.

glozow reviewed (commented):
Pushed for suggestions from @Xekyo @LarryRuane @bitschmidty (offline) - thanks!

glozow commented on _includes/specials/policy/en/05-dos.md:
```diff
@@ -0,0 +1,124 @@
+dollars per month in operational costs would very likely reduce the
+number of nodes on the network.
+
+Additionally, the role of a full node is wide open to threats. A node
```
Er this sounds a little bit too anthropomorphizing

bitschmidty commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,190 @@
+      the node _reacts_ to an incoming connection request.
+      Outbound nodes can be evicted, but that's done with an entirely
+      separate algorithm."
+  q0link="https://bitcoincore.reviews/27600#l-33"
```
```suggestion
  a0link="https://bitcoincore.reviews/27600#l-33"
```

harding replied:
Fixing and also adding test to automatically catch this wrong-parameter name typos in the future.

bitschmidty commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,190 @@
+      from eviction.
+      Without the PR, it would not return a peer if they all are excluded
+      (protected) from eviction."
+  q1link="https://bitcoincore.reviews/27600#l-70"
```
```suggestion
  a1link="https://bitcoincore.reviews/27600#l-70"
```

bitschmidty commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,190 @@
+      \"protected\" node might be evicted if necessary.)
+      However, as a result of discussion during the review club meeting,
+      the PR was later simplified such that this function is no longer modified."
+  q2link="https://bitcoincore.reviews/27600#l-126"
```
```suggestion
  a2link="https://bitcoincore.reviews/27600#l-126"
```

bitschmidty commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,190 @@
+      arguments, so there is no need for the function to be templated.
+      The PR's changes to this function were reverted, so it's still templated,
+      because changing this would be beyond the scope of the PR."
+  q3link="https://bitcoincore.reviews/27600#l-126"
```
```suggestion
  a3link="https://bitcoincore.reviews/27600#l-126"
```

bitschmidty commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,190 @@
+      that can be protected from eviction?"
+  a4="With and without the PR, the number would be 34 out of 40, assuming
+      they're not `noban` and inbound."
+  q4link="https://bitcoincore.reviews/27600#l-156"
```
```suggestion
  a4link="https://bitcoincore.reviews/27600#l-156"
```

bitschmidty commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,190 @@
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo], and
+[Bitcoin Inquisition][bitcoin inquisition repo].*
+
+- [Bitcoin Core #27501][] adds a `getprioritisedtransactions` RPC that
```
It might be nice to link to the PR Review Club coverage we had on this PR:

https://bitcoinops.org/en/newsletters/2023/05/10/#bitcoin-core-pr-review-club

LarryRuane commented:
Force-pushed to update the review club commit for @bitschmidty's comments, thanks. (Should I have made a new commit instead?)

harding commented:
@LarryRuane I prefer additional commits for just changes to the text.  Forced pushes are ok, though, and especially fine for cases where a commit message or metadata needs to be changed.

I made edits for suggested feedback, pushed a few smallish edits to @glozow's section (thanks!), reviewed the other contributributions, added the lede, added the releases/RCs, and added the topic entries.  Thanks everyone!

azuchi commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,200 @@
+## News
+
+- **Discussion about the taproot annex:** Joost Jager [posted][jager
+  annex] to the Bitcoin-Dev a request for a change in the Bitcoin Core
```
```suggestion
  annex] to the Bitcoin-Dev mailing list a request for a change in the Bitcoin Core
```

bitschmidty replied:
Good catch, pushed a correction. Thanks @azuchi !

joostjager commented on _posts/en/newsletters/2023-06-14-newsletter.md:
```diff
@@ -0,0 +1,200 @@
+    a problem at present.  Jager [replied][jager annex4] that he wants
+    to store signatures from ephemeral keys for a type of [vault][topic
+    vaults] that doesn't require a soft fork, and he  [suggested][jager
+    annex3] that some
```
One addition here could be that the annex seems to be the only way to create the vault and store the ephemeral signer signatures in one atomic operation in bitcoin currently.

bitschmidty replied:
@joostjager thanks for the feedback, but since we are so close to publishing, Im going to leave this out for now.

glozow commented on _includes/specials/policy/en/05-dos.md:
```diff
@@ -0,0 +1,108 @@
+nodes also do not verify any signatures before checking the feerate
+requirements. A non-mining node doesn’t ever receive fees – they are
+only paid to the miner who confirms the transaction – and some network
+bandwidth has already been used by the time the fee is checked.
```
Reworked this a little bit as I felt like I underexplained what "price" means.

bitschmidty commented:
Squashed and merged! Thank you @harding @glozow @LarryRuane @adamjonas @Xekyo @joostjager @azuchi for contributing and reviewing this week! 🚀
